### PR TITLE
Save build log

### PR DESCRIPTION
**Pull request by stappersg**
At least an attempt to save build log.
Actual need was saving ./test-suite.log

### CODE REVIEW

**Comment by melroy89** [comment]
It was pseudo code ;P

```


Check failure on line 23 in .github/workflows/linux.yaml
GitHub Actions / .github/workflows/linux.yaml

Invalid workflow file

You have an error in your yaml syntax on line 23
```

if you only want to save `test-suite.log` and the there is an error in the syntax anyways.  this..

File: .github/workflows/linux.yaml [hunk 1]
```diff
@@ -16,6 +16,11 @@ jobs:
           CC: ${{ matrix.compiler }}
         run: |
           test/test_build.sh alpine
+      - uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: log-file
+          path: *.log
```
Comment (melroy89) [inline .github/workflows/linux.yaml:23]:
```suggestion
          path: test-suite.log
```

File: .github/workflows/linux.yaml [hunk 2]
```diff
@@ -31,6 +36,11 @@ jobs:
           CC: ${{ matrix.compiler }}
         run: |
           test/test_build.sh debian
+      - uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: log-file
+          path: *.log
```
Comment (melroy89) [inline .github/workflows/linux.yaml:43]:
```suggestion
          path: ./test-suite.log
```

Comment (melroy89) [reply .github/workflows/linux.yaml:43]: Etc. etc. for the rest as well

**Comment by RICCIARDI-Adrien** [comment]
I also suggest to add different names for the artifacts, otherwise they will collide.

File: .github/workflows/linux.yaml [hunk 3]
```diff
@@ -16,6 +16,11 @@ jobs:
           CC: ${{ matrix.compiler }}
         run: |
           test/test_build.sh alpine
+      - uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: log-file
```
Comment (RICCIARDI-Adrien) [inline .github/workflows/linux.yaml:22]: Maybe use a unique name like `alpine-${{ matrix.compiler }}-log-file`.

**Comment by stappersg** [comment]
On Thu, Dec 05, 2024 at 01:30:19PM -0800, Melroy van den Berg wrote:
> > @@ -31,6 +36,11 @@ jobs:
>            CC: ${{ matrix.compiler }}
>          run: |
>            test/test_build.sh debian
> +      - uses: ***@***.***
> +        if: always()
> +        with:
> +          name: log-file
> +          path: *.log
> 
> ```suggestion
>           path: ./test-suite.log
> ```


***@***.***:~/src/githubforks/radvd/.github/workflows
$ git push --force
Enumerating objects: 9, done.
Counting objects: 100% (9/9), done.
Delta compression using up to 4 threads
Compressing objects: 100% (4/4), done.
Writing objects: 100% (5/5), 618 bytes | 618.00 KiB/s, done.
Total 5 (delta 3), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (3/3), completed with 3 local objects.
To github.com:radvd-project/radvd.git
 + ee5c3e7...d554454 save_ci_log -> save_ci_log (forced update)
***@***.***:~/src/githubforks/radvd/.github/workflows
$

Groeten
Geert Stappers
-- 
Silence is hard to parse

File: .github/workflows/linux.yaml [hunk 4]
```diff
@@ -31,6 +36,11 @@ jobs:
           CC: ${{ matrix.compiler }}
         run: |
           test/test_build.sh debian
+      - uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: log-file
```
Comment (RICCIARDI-Adrien) [inline .github/workflows/linux.yaml:42]: Maybe use a unique name like debian-${{ matrix.compiler }}-log-file.

Comment (RICCIARDI-Adrien) [reply .github/workflows/linux.yaml:42]: And so on for all the distros !

**Comment by stappersg** [comment]
On Thu, Dec 05, 2024 at 01:43:54PM -0800, Adrien RICCIARDI wrote:
> > @@ -31,6 +36,11 @@ jobs:
>            CC: ${{ matrix.compiler }}
>          run: |
>            test/test_build.sh debian
> +      - uses: ***@***.***
> +        if: always()
> +        with:
> +          name: log-file
> 
> Maybe use a unique name like debian-${{ matrix.compiler }}-log-file.

Work in progress ...

File: .github/workflows/linux.yaml [hunk 5]
```diff
@@ -80,3 +100,8 @@ jobs:
           CC: ${{ matrix.compiler }}
         run: |
           test/test_build.sh ubuntu
+      - uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: ubuntu-${{ matrix.compiler }}-log-file
```
Comment (RICCIARDI-Adrien) [inline .github/workflows/linux.yaml:106]: Sorry, I forgot that there were 2 Ubuntu versions !
I suggest `ubuntu-${{ matrix.ubuntu_version }}-${{ matrix.compiler }}-log-file`.

Comment (RICCIARDI-Adrien) [reply .github/workflows/linux.yaml:106]: You can even drop the initial `ubuntu`, using only `${{ matrix.ubuntu_version }}-${{ matrix.compiler }}-log-file`.

**Comment by stappersg** [comment]
On Thu, Dec 05, 2024 at 02:20:48PM -0800, Adrien RICCIARDI wrote:
> 
> You can even drop the initial `ubuntu`,
> using only `${{ matrix.ubuntu_version }}-${{ matrix.compiler }}-log-file`.

Nice. Done!


Groeten
Geert Stappers
-- 
Silence is hard to parse

**Review by RICCIARDI-Adrien — APPROVED** [review]
Artifacts are present in the workflow view, this looks good to me !

**Review by melroy89 — APPROVED** [review]

**Comment by Neustradamus** [comment]
@RICCIARDI-Adrien: What do you think?

**Comment by stappersg** [comment]
> @RICCIARDI-Adrien: What do you think?

I think that Neustradamus should try their/her/his privilege to approve merge requests for this project. 

image

**Comment by melroy89** [comment]
Sorry that my approval doesn't count 😜

**Comment by Neustradamus** [comment]
@stappersg: I know but I do not use my "power" ^^

**Review by RICCIARDI-Adrien — APPROVED** [review]

**Comment by RICCIARDI-Adrien** [comment]
@Neustradamus We all agree to merge this PR, what are we waiting for ?

**Review by melroy89 — APPROVED** [review]

**Comment by stappersg** [comment]
On Fri, Dec 06, 2024 at 01:26:09PM -0800, Neustradamus wrote:
> @stappersg: I know but I do not use my "power" ^^

As uncle Ben once said "With great power comes great responsebility."


@Neustradamus: Please take your responsebility.



Groeten
Geert Stappers
-- 
Silence is hard to parse

**Review by Neustradamus — APPROVED** [review]

**Comment by stappersg** [comment]
On Sat, Dec 07, 2024 at 09:07:59PM -0800, Neustradamus wrote:
> @Neustradamus approved this pull request.
> 

Thank you.

Please do what I did in #240, click on 'Merge pull request'.


Groeten
Geert Stappers
-- 
Silence is hard to parse

**Comment by melroy89** [comment]
> Please do what I did in #240, click on 'Merge pull request'.

That depends how you want to work together and what your WoW is. I normally like to merge my own PRs myself, to ensure a PR isn't merge too fast, causing possible regression for example.

**Comment by Neustradamus** [comment]
@robbat2: Good for you?

**Comment by stappersg** [comment]
State since 2024-12-28: Waiting on @Neustradamus 


Dear Neustradamus,

Please accept this invite for showing the _github universe_ that you have the power and the courage to merge merge requests.

One of your radvd project team members.

**Comment by stappersg** [comment]
On Sat, Dec 28, 2024 at 08:59:37AM -0800, Neustradamus wrote:
> Merged #237 into master.

Acknowledge.


Please surprise _github universe_ further with true action.





Regarding
Geert Stappers
Having the warm feeling that Neustradamus can more as poking people.
-- 
Silence is hard to parse
